packer/builder/azure/driver_restapi/builder.go

```diff
@@ -447,137 +442,45 @@ func (b *Builder) Cancel() {
 	}
 }
 
-func (b *Builder) validateAzureOptions(ui packer.Ui, state *multistep.BasicStateBag, reqManager *request.Manager) error {
+func (b *Builder) validateAzureOptions(ui packer.Ui, state *multistep.BasicStateBag) error {
 
 	var err error
 
 	// Check Storage account (& container)
 	ui.Message("Checking Storage Account...")
-
-	requestData := reqManager.CheckStorageAccountNameAvailability(b.config.StorageAccount)
-	resp, err := reqManager.Execute(requestData)
-
+	isStorageAvailable, _, err := storageservice.NewClient(b.client).IsAvailable(b.config.StorageAccount)
```

Comment:
`IsAvailable` is a weird name. we should've called it `CheckStorageAccountNameAvailability` on library side. To me availability mostly sounds about uptime. :)